en/guides/codablerouting.md

```diff
@@ -15,172 +15,263 @@ redirect_from:
+Kitura has two types of routing. Raw Routing, which works directly with Router Requests and Router Responses, and Codable Routing which works directly with Swift types, provided they conform to [Codable](https://developer.apple.com/documentation/swift/codable). 
 
-Codable Routing isn’t suitable for every use case and scenario, so Raw Routing is still available where you need the power and flexibility, but it is perfect for building REST APIs such as you might want to do to build a Backend For Frontend (BFF) for an iOS app.
+Codable Routing was made available in Kitura 2.0, before this release only Raw Routing was available. Kitura 2.0 supports both types of Routing, depending on what your application needs. You can implement both types of routing in your application as well. This offers a lot of flexibility on how you build your server. 
 
-## Building a Create API for POST requests
+For many, Codable Routing feels more 'natural' as you work directly with Swift types.  
+This guide shows you step by step how to set up Codable routes on a Kitura server. 
 
-Building a Codable Route for a POST API to implement storing a ToDo item requires only 4 lines of code. The first step is to register a handler for `POST` requests on “/todos”:
+## Let's build a Server!
 
-```swift
-router.post("/todos", handler: storeHandler)
-```
+### Getting started
+First we need a Kitura server to add our routes to. Kitura has a [command line interface](https://github.com/IBM-Swift/kitura-cli) which provides us with utilities for quickly creating a Kitura server.  
+> ![tip] You can find installation instructions for the command line interface in the [Getting Started](/{{ page.lang }}/starter/gettingstarted.html) section. 
 
-and then provide an implementation of the storeHandler:
+Once we've installed the Kitura command line interface we can open a terminal window and run the following:  
 
-```swift
-func storeHandler(todo: ToDo, completion: (ToDo?, RequestError?) -> Void ) -> Void {
-    todoStore.append(todo)
-    completion(todo, nil)
-}
-```
+1. Create a directory:  
+`mkdir ~/CodableRouting`
+2. Change to the new directory  
+`cd ~/CodableRouting`
+3. Create the Kitura server  
+`kitura init`
+
+`kitura init` downloads, creates and builds a starter Kitura application and will take a couple of minutes to complete. This includes best-practice implementations of health checking and monitoring as well as configuration files to allow easily deployment to a Docker container, a Kubernetes cluster, or the IBM Cloud.
 
-Here the store handler accepts a todo item of type `ToDo` and responds using an asynchronous completion handler. The handler then stores the received todo item in the todoStore and returns with the created todo item in the case of success, and an `Error` in the the case of failure.
+`kitura init` also creates an Xcode project for our server so we can open that now:  
```

Comment:
Can't we change this to a simpler - 
"Open the generated `CodableRouting.xcodeproj` project in Xcode:"